patroni/api.py

```diff
@@ -140,6 +152,22 @@ def do_GET(self, write_status_code_only=False):
             elif path in ('/async', '/asynchronous') and not is_synchronous:
                 status_code = replica_status_code
 
+        # check for user defined tags in query params
+        if status_code == 200:
+            qs_tag_prefix = "tag_"
+            qs_tags = {tag_name[len(qs_tag_prefix):]: tag_values[0]
+                       for tag_name, tag_values in self.path_query.items()
+                       if tag_name.startswith(qs_tag_prefix)}
+            for qs_key, qs_value in qs_tags.items():
+                qs_value = self.cast_query_param(qs_value)
+                instance_tag_value = patroni.tags.get(qs_key)
```

Comment:
IMO, since we are interested in comparing precise values it doesn't make much sense to cast str to other objects.
The opposite makes better sense.
I.e. we can cast `instance_tag_value` to a lower string if it is not None and not a string and use this value for comparison with query parameter value.

Reply:
The only reason I didn't do that was due to the simplicity of comparing floating point and integer values. Do you prefer the system to return 503 if 1.0 is compared to 1 or 200?

Reply:
done``

> > But since there was no documentation on the tags themselves I didn't consider doing so. I'll get to it.
> 
> There is: https://github.com/zalando/patroni/blob/master/docs/SETTINGS.rst#tags

I meant user defined tags. I'll add it to the doc